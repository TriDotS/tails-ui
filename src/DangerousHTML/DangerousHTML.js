import PropTypes from 'prop-types'
import React from 'react'

const DangerousHTML = ({ children }) => (
  <div dangerouslySetInnerHTML={{ __html: children }} />
)

DangerousHTML.propTypes = {
  children: PropTypes.string.isRequired
}

export default DangerousHTML
