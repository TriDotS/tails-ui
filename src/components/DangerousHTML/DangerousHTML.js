import PropTypes from 'prop-types'
import React from 'react'

const DangerousHTML = ({ children }) => (
  <div className="font-sans" dangerouslySetInnerHTML={{ __html: children }} />
)

DangerousHTML.propTypes = {
  children: PropTypes.string
}

export default DangerousHTML
