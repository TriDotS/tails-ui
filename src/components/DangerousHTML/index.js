import DangerousHTML from './DangerousHTML'

export default DangerousHTML
