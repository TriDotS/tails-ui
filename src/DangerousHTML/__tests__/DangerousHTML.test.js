import DangerousHTML from '../DangerousHTML'
import React from 'react'
import { render } from 'react-dom'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  render(<DangerousHTML />, div)
})

test('dangerous html snapshot', () => {
  const tree = renderer.create(<DangerousHTML />).toJSON()
  expect(tree).toMatchSnapshot()
})
