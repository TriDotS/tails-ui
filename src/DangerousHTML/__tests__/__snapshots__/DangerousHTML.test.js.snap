// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dangerous html snapshot 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "First Â· Second",
    }
  }
/>
`;
